doc/compiler/tril/TrilLanguageReference.md

```diff
@@ -0,0 +1,194 @@
+
+#### Properties
+
+* `name` _Optional_ Blocks can be named in order to target them with branches. 
+
+### Stores and loads
+
+Currently the creation of new symbol references in Tril is not supported.
+However, store and load opcodes provide two options for where the store / load
+can come from, which can be specified with one of the following two proprties: 
+
+* `parm` can specify the parameter from which a load can come. eg. `(iload
+  parm=2)` loads the third parameter from the signature. 
+* `temp` Stores to temps create new temporary variables that can later be
+  referenced. eg. `(istore temp="x" ...)`
+
+### Branches 
+
+Branches evaluate a condition then either jump to a target, or falls through. 
+
+#### Properties
+
+* `target` _Mandatory_ This is the block a successful computation of the
+  condition will continue executing in. 
+* `fallthrough` _Optional_ In some circumstances it is necessary to explicitly
+  list the fallthrough block of a branch to ensure the desired CFG is
+  constructed. There are two special values that may be used here: 
+   * `@none` indicates no fallthrough edge need be generated: this can be the
+     case for certain opcodes. 
+   * `@exit` falls through to the exit block of the CFG. 
+
+#### Example
+
+```
+(method ...
+   (block fallthrough="sub" name="start"
+      (ificmpge target="add"                 
+        ...) )
+   (block name="sub"
+      (ireturn
+         ...) )
+   (block name="add"
+      (ireturn                               
+        ... ) )
+```
+   
+### Commoning 
+
+Nodes can be given identifiers to allow creating commoned subtrees. To common a
+reference, on first compuation annotate the node with the `id` property. Then,
+where commoning is desired, replace the node with `(@id "...")` 
+
+```
+(imul id="multiplyResult"
+    ...) 
+(iadd
+  (@id "multiplyResult")
+  (@id "multiplyResult")
+```
+
+### Vectors 
+
+Vector support in Tril is very preliminary, but present. One of the most
+important pieces is the ability to annotate indirect vector loads and stores
+with the data type of the load. 
+
+This is required because the vector support in Testarossa infers the vector
+types from certain roots, as opposed to expressing the data types in the
+opcodes directly like other opcodes in Testarossa. 
+
+    (vloadi type=VectorDouble (aload parm=0))
+    
+####Properties
+
+* `type` _Mandatory_ The data type being loaded or stored. 
+
+
+
+### Calls
+_(Pending delivery as part of [#1910](https://github.com/eclipse/omr/pull/1910))_
+
+Call opcodes are supported with in Tril by providing extra annotation of the
+argument types and the function pointer to be invoked.
+
+    (icall address=0x1234 args=[Int32] ...) 
+
+####Properties
+
+* `address` _Mandatory_ The address of the function to be invoked by the call opcode
+* `args` _Optional_ The list of argument types passed to the call. 
+
+
+# Examples 
+
+## Incordec 
+
+Checked in as `fvtest/tril/examples/incordec/incordec.tril`
+
```

Comment:
Might be worth mentioning somewhere that line comments start with a `;`.

Reply:
done.